src/main/java/org/broadinstitute/hellbender/tools/coveragemodel/cachemanager/ComputableGraphStructure.java

```diff
@@ -0,0 +1,286 @@
+package org.broadinstitute.hellbender.tools.coveragemodel.cachemanager;
+
+import org.broadinstitute.hellbender.utils.Utils;
+
+import javax.annotation.Nonnull;
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+/**
+ * This class computes a number of useful structural properties for DAG specified by the
+ * immediate parents and immediate children of each node.
+ *
+ * @author Mehrtash Babadi &lt;mehrtash@broadinstitute.org&gt;
+ */
+public final class ComputableGraphStructure {
+
+    private final Set<String> nodeKeysSet;
+    private final Set<String> nodeTagsSet;
+    private final Map<String, Set<String>> allTagsMap;
+    private final Map<String, Set<String>> immediateDescendentsMap;
+    private final Map<String, Set<String>> immediateParentsMap;
+    private final Map<String, Set<String>> allDescendentsMap;
+    private final Map<String, Set<String>> allParentsMap;
+    private final Map<String, Integer> depthsMap;
+    private final Map<Integer, Set<String>> nodesByDepthMap;
+    private final Map<String, Set<String>> nodesByTagMap;
+    private final Map<String, List<String>> evalStrategyForNodeEvaluation;
+    private final Map<String, List<String>> evalStrategyForNodeMutation;
+    private final Map<String, List<String>> evalStrategyForTagEvaluation;
+    private final List<String> evalStrategyForAllNodes;
+
+    public ComputableGraphStructure(@Nonnull final Set<String> nodeKeysSet,
+                                    @Nonnull final Map<String, Set<String>> initialTagsMap,
+                                    @Nonnull final Map<String, Set<String>> immediateParentsMap,
```

Comment:
I dropped this explicit constructor as you suggested. We only have `ComputableGraphStructure(@Nonnull final Set<CacheNode> nodeSet)` now. Regarding dropping `immediate` -- I still like the distinction between `immediate` and `all` (or `full`, or something to that effect :D). Regarding `ancestor` vs. `parents` -- I have use `parents` all over the place. Is it fine with you if I keep it that way?